storage/SystemInfo/DevAndSys.cc

```diff
@@ -108,14 +108,14 @@ namespace storage
     }
 
 
-    template<>
-    int
-    File::get<int>() const
+    template<typename T>
+    T
+    File::get() const
     {
 	if (content.empty())
 	    ST_THROW(Exception("empty file " + path));
 
-	int ret;
+	T ret;
 	content.front() >> ret;
 
 	// TODO error checking
```

Comment:
Since we're touching this method...
According to 
- http://www.cplusplus.com/reference/istream/istream/operator%3E%3E/ 
- http://www.cplusplus.com/reference/ios/ios/rdstate/

we can check for an error with `if (!content.good())`.

Given the `ST_THROW` above, I think another one here would be fine.

Reply:
@mvidner but content according to header file, it is vector<string> so not istream, so I worruy iostream methods won't work? I think it is converted by some boost string extension? For me searching for that operand in boost documentation does not end well ( for me boost documentation is a bit hard to read and its header files even harder :)

Reply:
Yes,  ```content``` is a ```vector``` not a ```stream```. Martin likely meant to use ```good()``` on a ```stream``` but in the code is no ```stream``` visible. The ```operator>>``` from StorageTmpl.h is used and hides the ```stream```. So the few lines from ```operator>>``` could be copied here and then error checking can be added,